### ISSUE TITLE
OSX Yosemite: Command-Q (quit) doesn't quit Light Table, it only closes the current workspace.

### ISSUE THREAD

**Issue by JohnPostlethwait** [LightTable/LightTable#1826]
Quit should quit the application. I know of no other OSX app that quits only the open window, but all other ones have to be quit independently.

**Comment by cldwalker** [comment]
I agree Command-Q/Quit Light Table cmd should close all workspaces. We should address this after atom-shell has landed since it touches that code.

**Comment by kenny-evitt** [comment]
This seems to be fixed in the *atom-shell* branch.

**Comment by cldwalker** [comment]
On osx 10.10 with latest LT this can be reproduced by opening two windows and pressing Command-Q. Only the current window closes when all windows and the whole application is expect to be closed. This may only be an osx expectation. Will be adding this to the roadmap

**Comment by kenny-evitt** [comment]
Maybe I misunderstood – or maybe this really was fixed (or seemed to have been) when I tried it long ago (and presumably on a Mac).

But I just reproduced this on a Windows 7 computer [T]. In an open LT window, I opened the *File* menu and picked *New window*. In the new LT window I opened the *File* menu and picked *Quit Light Table*. Only the new window was closed.

In my experience, as a long-time Windows user, this is *mostly*-expected. Some applications work as a collection of windows whereby quitting closes all of them. But lots of other applications provide means of 'spawning' a new instance of the application in a new window and the new instance behaves as a separate instance of the application – closing or even quitting one doesn't close or quit all of the others.

My experience with Mac applications is relatively more limited, but I've found it much rarer to be able to create new 'windows' for applications mostly contained in a single window. I just kinda figured that the *New window* was creating an entirely separate instance of the LT app and thus the quit not closing all windows is not surprising.

I'm not aware of any corresponding conventions or expectations with regard to Linux GUI applications.

**Comment by sbauer322** [comment]
Adding a Mac label since this is unexpected behavior in that space.
